modules/planning/scenarios/scenario_manager.cc

```diff
@@ -910,10 +910,36 @@ void ScenarioManager::UpdatePlanningContextTrafficLightScenario(
 // update: pull_over status in PlanningContext
 void ScenarioManager::UpdatePlanningContextPullOverScenario(
     const Frame& frame, const ScenarioConfig::ScenarioType& scenario_type) {
-  if (scenario_type != ScenarioConfig::PULL_OVER) {
-    PlanningContext::Instance()->mutable_planning_status()->clear_pull_over();
+  if (scenario_type == ScenarioConfig::PULL_OVER) {
     return;
   }
+
+  const auto& pull_over_status =
+      PlanningContext::Instance()->planning_status().pull_over();
+  if (pull_over_status.has_x() && pull_over_status.has_y()) {
+    const auto& routing = frame.local_view().routing;
+    if (routing->routing_request().waypoint_size() >= 2) {
+      // keep pull-over stop fence if destination not changed
+      const auto& reference_line_info = frame.reference_line_info().front();
+      const auto& reference_line = reference_line_info.reference_line();
+
+      common::SLPoint dest_sl;
+      const auto& routing_end =
+          *(routing->routing_request().waypoint().rbegin());
+      reference_line.XYToSL({routing_end.pose().x(), routing_end.pose().y()},
+                            &dest_sl);
+
+      common::SLPoint pull_over_sl;
+      reference_line.XYToSL({pull_over_status.x(), pull_over_status.y()},
+                            &pull_over_sl);
+
+      constexpr double kDestMaxDelta = 30.0;  // meter
```

Comment:
@jmtao In PathBoundDecider there is also a `pull_over_destination_to_pathend_buffer` in us as in https://github.com/ApolloAuto/apollo/blob/abe92a3111dda9c7e8fd903997424d1161115e31/modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc#L439 , will these be of conflict or is it better if we can merge these two?